GUI/PyQt/Qt_main.py

```diff
@@ -243,142 +77,175 @@ def resample(self, image, scan, new_spacing=[1, 1, 1]):
         return image, new_spacing
 
     def load_data(self):
-        self.slider.setValue(1)
-        self.PathDicom = QtWidgets.QFileDialog.getExistingDirectory(self, "open file", "C:/Users/hansw/Videos/artefacts")  # root directory
-        if self.PathDicom:
-            print(self.PathDicom)
-            files = sorted([os.path.join(self.PathDicom, file) for file in os.listdir(self.PathDicom)], key=os.path.getctime)
-            datasets = [dicom.read_file(f) \
-                        for f in files]
-            try:
-                self.voxel_ndarray, pixel_space = dicom_numpy.combine_slices(datasets)
-            except dicom_numpy.DicomImportException:
-                raise
-
-        self.sscan = self.load_scan(self.PathDicom)
-        if self.sscan:
-            self.simage = np.stack([s.pixel_array for s in self.sscan])
-            self.svoxel, spacing = self.resample(self.simage, self.sscan, [1, 1, 1])
-            self.svoxel = np.swapaxes(self.svoxel, 0, 2)
-            self.ax1 = self.canvas.figure.add_subplot(111)
-
-    def updatea(self):
-        if self.voxel_ndarray == []:
-            pass
+        if self.gridson == False:
+            QtWidgets.QMessageBox.information(self, 'Warning', 'Grids needed!')
         else:
-            self.slider.valueChanged.disconnect()
-            self.slices = self.voxel_ndarray.shape[2]
-            self.slider.setRange(1, self.slices)
-            self.slider.setValue(1)
-            self.slider.valueChanged.connect(self.View_A)
-            # spinbox
-            #self.valueset.setRange(1, self.slices)
-            #self.valueset.valueChanged.connect(self.slice_update)
-            #self.valueset.setValue(1)
-
-
-    def View_A(self):
-        if self.voxel_ndarray == []:
+            self.PathDicom = QtWidgets.QFileDialog.getExistingDirectory(self, "open file", "C:/Users/hansw/Videos/artefacts")  # root directory

```

Comment:
The path to the dicom directory can be obtained via 
```
with open('config' + os.sep + 'param.yml', 'r') as ymlfile:
cfg = yaml.safe_load(ymlfile)
dbinfo = DatabaseInfo(cfg['MRdatabase'],cfg['subdirs'])
```

And then `dbinfo.sPathIn`
or 
`os.path.join(dbinfo.sPathIn, pat, dbinfo.sSubDirs[1], seq)`
with `pat = dbinfo.lPats[i]` and `seq = cfg['selectedDatabase']['dataref']` or `seq= cfg['selectedDatabase']['dataart']`
is what you are looking for

Reply:
I add this into my code, now it works. But this setting only fit for the path in our labor. If the user want to use the GUI on other computer, they must make change in the 27th line of DatabaseInfo.py :

self.sPathIn = '/med_data/ImageSimilarity/Databases'+ os.sep + self.sDatabase + os.sep + self.sSubDirs[0]

 '/med_data/ImageSimilarity/Databases' needs to be changed to the path which contains the database.